frame/collective/src/lib.rs

```diff
@@ -182,72 +343,224 @@ decl_module! {
 
 		/// Set the collective's membership.
 		///
-		/// - `new_members`: The new member list. Be nice to the chain and
-		//	provide it sorted.
+		/// - `new_members`: The new member list. Be nice to the chain and provide it sorted.
 		/// - `prime`: The prime member whose vote sets the default.
+		/// - `old_count`: The upper bound for the previous number of members in storage.
+		///                Used for weight estimation.
 		///
 		/// Requires root origin.
-		#[weight = (100_000_000, DispatchClass::Operational)]
-		fn set_members(origin, new_members: Vec<T::AccountId>, prime: Option<T::AccountId>) {
+		///
+		/// NOTE: Does not enforce the expected `MAX_MEMBERS` limit on the amount of members, but
+		///       the weight estimations rely on it to estimate dispatchable weight.
+		///
+		/// # <weight>
+		/// ## Weight
+		/// - `O(MP + N)` where:
+		///   - `M` old-members-count (code- and governance-bounded)
+		///   - `N` new-members-count (code- and governance-bounded)
+		///   - `P` proposals-count (code-bounded)
+		/// - DB:
+		///   - 1 storage mutation (codec `O(M)` read, `O(N)` write) for reading and writing the members
+		///   - 1 storage read (codec `O(P)`) for reading the proposals
+		///   - `P` storage mutations (codec `O(M)`) for updating the votes for each proposal
+		///   - 1 storage write (codec `O(1)`) for deleting the old `prime` and setting the new one
+		/// # </weight>
+		#[weight = (
+			weight_for::set_members::<T, I>(
+				*old_count as Weight, // M
```

Comment:
I feel safer to use
```suggestion
				(*old_count).into(), // M
```
in case MemberCount type is bigger than Weight type `as` will wrap around (thought that should never happen)
but anyway fine.